src/Compiler/Symbols/FSharpDiagnostic.fs

```diff
@@ -236,22 +235,9 @@ type FSharpDiagnostic(m: range, severity: FSharpDiagnosticSeverity, message: str
              | _ -> diagnostic.FormatCore(flatErrors, suggestNames)
 
         let errorNum = diagnostic.Number
+        let m = match diagnostic.Range with Some m -> m | None -> range.Zero
```

Comment:
I'm not sure what the idea behind the `Zero` range is.
There are also `range0`, `rangeStartup`, and `rangeCmdArgs`. Do we have any guarantees about what indices these filenames get?

It seems to me that `Zero` is implicitly linked to the filename at index `0`, but what if it happens to be a real file, like the `Program.fs`? Can it also mean a wrong file when there are multiple projects?

Maybe we could reserve some special indices? Or use an invalid index like `-1` to distinguish them?
Should we rename `Zero` to something like `Invalid` and add helper members like `IsValid`? That seems to be the best option to me, if feasible.

Reply:
OTOH `Range.Zero` seems to refer explicitly to file with index 0 whichever it is (?)

Reply:
> OTOH Range.Zero seems to refer explicitly to file with index 0 whichever it is (?)

Yes, this is what puzzles me. We need to guarantee this is not a real file in a project.

Reply:
I used Range.Zero because that was used before in all the places that called `CreateFromException` without having a proper fallback range. (5 places in this PR). But that might not have been the best choice.
Let me look into it.